File: baekjoon/15650.cpp
#include <iostream>
#include <vector>
using namespace std;

int N, M;
bool visit[9]{};
vector<int> v;  // arr[depth]

void dfs(int k, int level) {
    if(level == M) {
        for(int i = 0; i < v.size(); i++) {
            cout << v[i] << " ";
        }
        cout << "\n";
        return;
    }
    
    for(int i = 1; i <= N; i++) {
        
        if(visit[i]) continue;
        if(i <= k) continue;
        
        v.push_back(i);
        visit[i] = true;
        dfs(i, level + 1);
        v.pop_back();
        visit[i] = false;
    }
}

int main()
{
    cin >> N >> M;
    dfs(0, 0);
}

File: baekjoon/15650.py
n, m = map(int, input().split())
arr = [0] * m

def recur(val, level):
    if level == m:
        print(*arr)
        return 
                
    for i in range(val, n+1):
        arr[level] = i
        recur(i+1, level+1)

recur(1, 0)

